llama_stack/providers/inline/agents/meta_reference/agent_instance.py

```diff
@@ -916,12 +925,12 @@ async def execute_tool_call_maybe(
     messages: List[CompletionMessage],
     toolgroup_args: Dict[str, Dict[str, Any]],
     tool_to_group: Dict[str, str],
-) -> List[ToolResponseMessage]:
+) -> Tuple[ToolCall, ToolInvocationResult]:
```

Comment:
feels odd to return a "tool_call" from this method. I think this method needs to be broken up because it is just doing too much. can you hoist up the tool_call extraction into the calling code and just have this method operate on the tool_call, and just do some tool group fiddling and invoking? so this method should be `f: tool_call -> tool_result`